src/FSharp.Stats/Integration/Integration.fs

```diff
@@ -1,5 +1,118 @@
 namespace FSharp.Stats.Integration
 
+open FSharp.Stats
+open FSharpAux
+
+/// Approximation methods for definitive integral estimation
+type NumericalIntegrationMethod =
+    | LeftEndpoint
+    | RightEndpoint
+    | Midpoint
+    | Trapezoidal
+    | Simpson
+
+    static member getIntervalIntegrals = function 
+        | LeftEndpoint -> fun (f: (float -> float)) leftBoundaries partitionSize -> leftBoundaries |> Seq.map (fun boundary -> f(boundary) * partitionSize)
+        | RightEndpoint -> fun (f: (float -> float)) leftBoundaries partitionSize -> leftBoundaries |> Seq.map (fun boundary -> f(boundary + partitionSize) * partitionSize)
+        | Midpoint -> fun (f: (float -> float)) leftBoundaries partitionSize -> leftBoundaries |> Seq.map (fun boundary -> f(boundary + (partitionSize / 2.)) * partitionSize)
+        | Trapezoidal -> fun (f: (float -> float)) leftBoundaries partitionSize -> leftBoundaries |> Seq.map (fun boundary -> ((f(boundary) + f(boundary + partitionSize)) * partitionSize) / 2.)
+        | Simpson -> fun (f: (float -> float)) leftBoundaries partitionSize -> leftBoundaries |> Seq.map (fun boundary -> ((f(boundary) + 4.0 * f(boundary + partitionSize / 2.0) + f(boundary + partitionSize)) * partitionSize) / 6.0)
+
+
+    static member integrateFunction (method: NumericalIntegrationMethod) = 
+         fun (f: (float -> float)) leftBoundaries partitionSize -> NumericalIntegrationMethod.getIntervalIntegrals method f leftBoundaries partitionSize |> Seq.sum
+        
+
+    static member getObservationIntegrals = function
+        | LeftEndpoint -> fun (observations: (float*float) []) ->
+            let xVals, yVals = observations |> Array.unzip
+            xVals
+            |> Array.mapi (fun i x ->
+                if i = xVals.Length - 1 then 
+                    0.
+                else 
+                    let rectWidth = xVals[i+1] - x
+                    (rectWidth*yVals[i])
+            )
+        | RightEndpoint -> fun (observations: (float*float) []) ->
+            let xVals, yVals = observations |> Array.unzip
+            xVals
+            |> Array.mapi (fun i x ->
+                if i = 0 then 
+                    0. 
+                else 
+                    let rectWidth = x - xVals[i-1]
+                    (rectWidth*yVals[i])
+            )
+        | Midpoint -> fun (observations: (float*float) []) ->
```

Comment:
I think there is a problem with the midpoint strategy: 
- As you can see the performed calculations are equal to the Trapezoidal method. The result would always be the same. 
- Since the midpoint rule requires a function to calculate the true value of (a+b)/2 it can **only be applied if a function is given**. For `float*float` inputs this is impossible and therefore should be either removed or midpoint should ask for a `(f: float -> float)` input with interval rather than a `(float*float) []`

![image](https://user-images.githubusercontent.com/28917670/162166275-5588c931-d0f8-44d5-80a6-ab8373ab95eb.png)

Reply:
this is only the case for midpoint rule definite integrals for observations. i added a hint in the xml docs fr the midpoint rule.